opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -1121,6 +1167,88 @@ namespace detail {
+                                                              SolutionState& state,
+                                                              WellState& well_state)
+    {
+        V aliveWells;
+        const int np = wells().number_of_phases;
+        std::vector<ADB> cq_s(np, ADB::null());
+
+        int          it  = 0;
+        std::vector<V> vars0;
+        //bhp and Q for the wells
+        vars0.reserve(2);
+        variableWellStateInitials(well_state,vars0);
+        std::vector<ADB> vars = ADB::variables(vars0);
+        std::vector<int> indices = variableWellsStateIndices();
+        SolutionState state0 = state;
+        asImpl().makeConstantState(state0);
+        SolutionState wellSolutionState = state0;
+        variableStateExtractWellsVars(indices,vars,wellSolutionState);
+        std::vector<ADB> mob_perfcells_const(np, ADB::null());
+        std::vector<ADB> b_perfcells_const(np, ADB::null());
+        for (int phase = 0; phase < np; ++phase) {
+            mob_perfcells_const[phase] = ADB::constant(mob_perfcells[phase].value());
+            b_perfcells_const[phase] = ADB::constant(b_perfcells[phase].value());
+        }
+        asImpl().addWellEq(wellSolutionState, well_state, mob_perfcells_const, b_perfcells_const, aliveWells,cq_s);
+        addWellControlEq(wellSolutionState, well_state, aliveWells);
+        bool converged = getWellConvergence(it);
+        while ( (!converged && (it < 15))) {
+
+            std::vector<ADB> eqs;
+            eqs.reserve(2);
+            eqs.push_back(residual_.well_flux_eq);
+            eqs.push_back(residual_.well_eq);
+            ADB total_residual = vertcatCollapseJacs(eqs);
+            const std::vector<M>& Jn = total_residual.derivative();
+            const Eigen::SparseLU< M > solver(Jn[0]);
+            const Eigen::VectorXd& dx = solver.solve(total_residual.value().matrix());
+            const int numeq = well_state.numWells()*(well_state.numPhases()+1);
+            V dx_V = V(numeq);
+            std::copy_n(dx.data(),numeq, dx_V.data());
+            updateWellState(dx_V,well_state);
+            updateWellControls(well_state);
+            //bhp and Q for the wells
+            vars0.clear();
+            variableWellStateInitials(well_state,vars0);
+            vars = ADB::variables(vars0);
+            wellSolutionState = state0;
+            variableStateExtractWellsVars(indices,vars,wellSolutionState);
+            asImpl().addWellEq(wellSolutionState, well_state, mob_perfcells_const, b_perfcells_const, aliveWells,cq_s);
+            addWellControlEq(wellSolutionState, well_state, aliveWells);
+            it++;
+            converged = getWellConvergence(it);
+        }
+        if (converged) {
+            std::cout << "well converged iter: " << it << std::endl;
+            const int nw = wells().number_of_wells;
+            {
+                // We will set the bhp primary variable to the new ones,
```

Comment:
With using 
```
state.bhp = vars[1];
state.qs = vars[0];
```
The following problem was triggered. Have not done more investigation. 
```
Program received signal SIGSEGV, Segmentation fault.
0x0000000000e912a2 in Eigen::SparseMatrix<double, 0, int>::cols (this=0xc0) at /usr/local/include/eigen3/Eigen/src/SparseCore/SparseMatrix.h:121
121	    inline Index cols() const { return IsRowMajor ? m_innerSize : m_outerSize; }
(gdb) backtrace 
#0  0x0000000000e912a2 in Eigen::SparseMatrix<double, 0, int>::cols (this=0xc0) at /usr/local/include/eigen3/Eigen/src/SparseCore/SparseMatrix.h:121
#1  0x0000000000f5d7e2 in Eigen::ProductBase<Eigen::SparseTimeDenseProduct<Eigen::SparseMatrix<double, 0, int>, Eigen::Matrix<double, -1, 1, 0, -1, 1> >, Eigen::SparseMatrix<double, 0, int>, Eigen::Matrix<double, -1, 1, 0, -1, 1> >::ProductBase (this=0x7fffffffc060, a_lhs=..., 
    a_rhs=...) at /usr/local/include/eigen3/Eigen/src/Core/ProductBase.h:93
#2  0x0000000000f593f1 in Eigen::SparseTimeDenseProduct<Eigen::SparseMatrix<double, 0, int>, Eigen::Matrix<double, -1, 1, 0, -1, 1> >::SparseTimeDenseProduct (this=0x7fffffffc060, lhs=..., rhs=...) at /usr/local/include/eigen3/Eigen/src/SparseCore/SparseDenseProduct.h:245
#3  0x0000000000f55637 in Eigen::SparseMatrixBase<Eigen::SparseMatrix<double, 0, int> >::operator*<Eigen::Matrix<double, -1, 1, 0, -1, 1> > (this=0xc0, other=...) at /usr/local/include/eigen3/Eigen/src/SparseCore/SparseDenseProduct.h:296
#4  0x0000000000f4ebcc in Opm::(anonymous namespace)::eliminateVariable (eqs=..., n=3) at /home/kaib/OPM/debug/opm-autodiff/opm/autodiff/NewtonIterationBlackoilCPR.cpp:283
#5  0x0000000000f4de57 in Opm::NewtonIterationBlackoilCPR::computeNewtonIncrement (this=0x76a6890, residual=...) at /home/kaib/OPM/debug/opm-autodiff/opm/autodiff/NewtonIterationBlackoilCPR.cpp:144
#6  0x0000000000eb836d in Opm::BlackoilModelBase<UnstructuredGrid, Opm::BlackoilModel<UnstructuredGrid> >::solveJacobianSystem (this=0x8f33060) at /home/kaib/OPM/debug/opm-autodiff/opm/autodiff/BlackoilModelBase_impl.hpp:1385
#7  0x0000000000ea5d73 in Opm::NewtonSolver<Opm::BlackoilModel<UnstructuredGrid> >::step (this=0x7885410, dt=69120.000100000005, reservoir_state=..., well_state=...) at /home/kaib/OPM/debug/opm-autodiff/opm/autodiff/NewtonSolver_impl.hpp:85
#8  0x0000000000eb5875 in Opm::AdaptiveTimeStepping::stepImpl<Opm::NewtonSolver<Opm::BlackoilModel<UnstructuredGrid> >, Opm::BlackoilState, Opm::WellStateFullyImplicitBlackoil> (this=0x7882ad0, simulatorTimer=..., solver=..., state=..., well_state=..., outputWriter=0x7fffffffdbe0)
    at /home/kaib/OPM/debug/opm-core/opm/core/simulator/AdaptiveTimeStepping_impl.hpp:138
#9  0x0000000000ea5b6e in Opm::AdaptiveTimeStepping::step<Opm::NewtonSolver<Opm::BlackoilModel<UnstructuredGrid> >, Opm::BlackoilState, Opm::WellStateFullyImplicitBlackoil> (this=0x7882ad0, simulatorTimer=..., solver=..., state=..., well_state=..., outputWriter=...)
    at /home/kaib/OPM/debug/opm-core/opm/core/simulator/AdaptiveTimeStepping_impl.hpp:90
```